ui/jetbrains/src/main/java/com/neueda/jetbrains/plugin/graphdb/jetbrains/ui/datasource/interactions/DataSourceDialog.java

```diff
@@ -35,51 +41,93 @@ public boolean go() {
         return showAndGet();
     }
 
-    public void validationPopup() {
+    private void showLoading(JButton testConnectionButton, JPanel loadingPanel, AsyncProcessIcon loadingIcon) {
```

Comment:
Can all the parameters be converted into the DataSourceDialog instance fields?

Reply:
I move it to upper level. Not its abstract and implementation is in Neo4j/Gremlin Dialogs